.github/workflows/deploy.yml

```diff
@@ -0,0 +1,57 @@
+name: Test and Deploy to GitHub Pages
+
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches: ["main"]
+
```

Comment:
Could be nice if the tests ran on PRs and push to main, and deploy ran on push to main, but it's not a big deal.

Reply:
Looks great, thanks!